tests/Unit/Parallel/Test_PupStlCpp17.cpp

```diff
@@ -19,10 +39,21 @@ SPECTRE_TEST_CASE("Unit.Serialization.PupStlCpp17", "[Serialization][Unit]") {
     CHECK_FALSE(t_not_set_deserialized.has_value());
 
     const std::optional<int> t_set{10};
-    REQUIRE(t_set.value());
+    REQUIRE(t_set.value() == 10);
     const auto t_set_deserialized = serialize_and_deserialize(t_set);
     REQUIRE(t_set_deserialized.value());
     CHECK(*t_set_deserialized == 10);
+
+    // Test that we can handle a non-copyable and non-movable class. This occurs
+    // when you have a hashtable where the mapped type is a
+    // std::optional<NonCopyable>
+    const std::optional<LocalNoCopyMove> t_nc{10};
+    REQUIRE(t_nc.value().t == 10);
+    std::optional<LocalNoCopyMove> t_nc_deserialized{};
+    const std::vector<char> serialized = serialize(t_nc);
```

Comment:
There's also a not_null `serialize_and_deserialize` in test helpers now.

Reply:
Okay, switched to that, thanks!